### Q1. What are the three measures of central tendency?
    
    The three measures of central tendency are :
                                                Mean
                                                Median
                                                Mode

## Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?
        
    1) Mean :
    ---------
        The mean is the arithmetic average, and it is probably the measure of central tendency that we are most familiar. Calculating the mean is very simple. We just add up all of the values and divide by the number of observations in your dataset.
        The calculation of the mean incorporates all values in the data. If we change any value, the mean changes. However, the mean doesn’t always locate the center of the data accurately.
    
    2) Median : 
    ------------
        The median is the middle value. It is the value that splits the dataset in half, making it a natural measure of central tendency.
        To find the median, order our data from smallest to largest, and then find the data point that has an equal number of values above it and below it. 
        The method for locating the median varies slightly depending on whether our dataset has an even or odd number of values. When there is an even number of values, you count in to the two innermost values and then take the average.
    
    3) Mode :
    ----------
        The mode is the value that occurs the most frequently in our data set, making it a different type of measure of central tendency than the mean or median.

        To find the mode, sort the values in our dataset by numeric values or by categories. Then identify the value that occurs most often.

### Q3. Measure the three measures of central tendency for the given height data:
    
    [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [58]:
import numpy as np
from scipy import stats

lst = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

print(f"Mean   : {np.mean(lst)}")
print(f"Median : {np.median(lst)}")
print(f"Mode   : {stats.mode(lst, keepdims=True)}")

Mean   : 177.01875
Median : 177.0
Mode   : ModeResult(mode=array([177.]), count=array([3]))


### Q4. Find the standard deviation for the given data:
    [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [59]:
import pandas as pd

lst1 = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
df = pd.DataFrame(lst1)
print(f"Standard Deviation : {df.std(axis=0).values}")

Standard Deviation : [1.84723893]


## Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.
     
     The range, variance, and standard deviation are all measures of dispersion that provide information about how spread out the data values are.
     The range is a simple measure that is sensitive to outliers, while the variance and standard deviation are more robust measures that take into account all the values in the dataset. 
     The standard deviation is particularly useful for describing the spread of data that follows a normal distribution.

In [60]:
import numpy as np

scores = np.array([70, 72, 73, 75, 76, 78, 80, 81, 82, 83, 84, 85, 86, 88, 89, 90, 92, 94, 96, 98])

print(f"Range              : {np.ptp(scores)}")
print(f"Variance           : {np.var(scores):.4f}")
print(f"Standard Deviation : {np.std(scores):.4f}")


Range              : 28
Variance           : 62.7400
Standard Deviation : 7.9209


The range is the difference between the maximum and minimum values in the dataset. In this case, the range is 28.

The variance is the average of the squared differences of each value from the mean. It measures how much the data points are spread out from the mean. In this case, the variance is: 62.7400

The standard deviation is the square root of the variance. It is a measure of how spread out the data is from the mean, and is commonly used to describe the variability in a dataset. In this case, the standard deviation is: 7.9209

### Q6. What is a Venn diagram?
    A Venn diagram uses overlapping circles or other shapes to illustrate the logical relationships between two or more sets of items.

### Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
    (i) A B
    (ii) A ⋃ B

In [61]:
import numpy as np

A = np.array([2,3,4,5,6,7])
B = np.array([0,2,6,8,10])

print(f"Intersection of A & B sets : {np.intersect1d(A, B)}")
print(f"Union of A & B sets        : {np.union1d(A, B)}")

Intersection of A & B sets : [2 6]
Union of A & B sets        : [ 0  2  3  4  5  6  7  8 10]


### Q8. What do you understand about skewness in data?

    Skewness is a measurement of the distortion of symmetrical distribution or asymmetry in a data set. 
    
    Skewness is demonstrated on a bell curve when data points are not distributed symmetrically to the left and right sides of the median on a bell curve.

### Q9. If a data is right skewed then what will be the position of median with respect to mean?
    
    When the distribution is skewed to the right, the mean is often greater than the median.

### Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?
        
        Covariance and correlation are two measures of the relationship between two variables in a dataset. While both measures are used to describe the strength and direction of the relationship between variables.
        
        Covariance shows us how the two variables differ, whereas correlation shows us how the two variables are related.
        
         covariance and correlation are used to determine the strength and direction of the relationship between two variables. These measures are often used in regression analysis, where the goal is to predict the value of one variable (the dependent variable) based on the value of one or more other variables (the independent variables). Correlation is also used to determine the degree of association between two variables in other statistical analyses, such as hypothesis testing or factor analysis.
    

### Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.
    
    The formula for calculating the sample mean (also known as the sample average) is:

        sample mean = sum of observations / number of observations

In [62]:
import numpy as np

arr = np.array([3, 5, 8, 2, 9])
print(f"Mean of array : {np.mean(arr)}")

Mean of array : 5.4


### Q12. For a normal distribution data what is the relationship between its measure of central tendency?
    
    For a normal distribution, the mean, median, and mode are all equal. In other words, the measure of central tendency for a normal distribution is the same, regardless of whether we use the mean, median, or mode to describe it.
    
    This is because a normal distribution is symmetric around its mean. The mean is the arithmetic average of all the data points, the median is the middle value of the dataset when it is ordered from smallest to largest, and the mode is the most common value in the dataset. Since the normal distribution is symmetric, the mean, median, and mode all lie at the same point in the distribution.

### Q13. How is covariance different from correlation?

    Covariance and correlation are two measures that describe the relationship between two variables. While both measures are used to describe the direction and strength of the relationship, there are some differences between them:

    Standardization: Covariance is not standardized, while correlation is standardized. This means that the magnitude of the covariance depends on the scale of the variables being measured, while the correlation coefficient always ranges from -1 to 1.

    Interpretation: Covariance can take on any value, including negative values, while correlation ranges from -1 (perfect negative correlation) to 1 (perfect positive correlation). Correlation values of 0 indicate no linear relationship between the variables.

    Units: Covariance is measured in units that are the product of the units of the two variables being measured, while correlation is unitless.

    Sensitivity to outliers: Covariance is sensitive to outliers in the data, while correlation is more robust to outliers.

    In summary, covariance and correlation are both measures of the relationship between two variables, but covariance is not standardized and is affected by the units of the variables being measured, while correlation is standardized, unitless, and more robust to outliers.

### Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.
    
    Outliers are extreme values in a dataset that are much higher or lower than the rest of the data. Outliers can have a significant impact on measures of central tendency and dispersion, as they can pull the mean, median, mode, and measures of spread away from the typical values of the dataset.

In [63]:
import numpy as np
from scipy import stats

arr1 = np.array([78, 89, 92, 84, 96, 90, 85, 87, 90])
arr2 = np.array([78, 89, 92, 84, 96, 90, 85, 87, 90, 158])

print(f" without outliers : Mean : {np.mean(arr1):.4f}, Median {np.median(arr1)}, Mode {stats.mode(arr1, keepdims=True)}, Variance {np.var(arr1):.4f}, Standard Deviation {np.std(arr1):.4f}")
print(f" with outliers :    Mean : {np.mean(arr2):.4f}, Median {np.median(arr2)}, Mode {stats.mode(arr2, keepdims=True)}, Variance {np.var(arr2):.4f}, Standard Deviation {np.std(arr2):.4f}")

 without outliers : Mean : 87.8889, Median 89.0, Mode ModeResult(mode=array([90]), count=array([2])), Variance 23.8765, Standard Deviation 4.8864
 with outliers :    Mean : 94.9000, Median 89.5, Mode ModeResult(mode=array([90]), count=array([2])), Variance 463.8900, Standard Deviation 21.5381


As we can see, the outlier has had a significant impact on the measures of central tendency and dispersion. The mean, the variance and the standard deviation have all increased significantly. This is because the outlier is so far from the rest of the data that it pulls the measures of central tendency and dispersion towards it.

This example demonstrates the importance of identifying and dealing with outliers in a dataset, as they can significantly affect the results of statistical analysis.